execution-core/src/stake.rs

```diff
@@ -197,8 +197,41 @@ impl Withdraw {
 pub struct StakeEvent {
     /// Keys associated to the event.
     pub keys: StakeKeys,
-    /// Value of the relevant operation, be it `stake`, `unstake`,`withdraw`
+    /// Effective value of the relevant operation, be it `stake`,
+    /// `unstake`,`withdraw`
     pub value: u64,
+    /// The locked amount involved in the operation (e.g., for `stake` or
+    /// `unstake`). Defaults to zero for operations that do not involve
+    /// locking.
+    pub locked: u64,
+}
+
+impl StakeEvent {
+    /// Creates a new `StakeEvent` with the specified keys and value.
+    ///
+    /// ### Parameters
+    /// - `keys`: The keys associated with the stake event.
+    /// - `value`: The effective value of the operation (e.g., `stake`,
+    ///   `unstake`, `withdraw`).
+    ///
+    /// The `locked` amount is initialized to zero by default.
+    #[must_use]
+    pub fn new(keys: StakeKeys, value: u64) -> Self {
+        Self {
+            keys,
+            value,
+            locked: 0,
+        }
+    }
+    /// Sets the locked amount for the `StakeEvent`.
+    ///
+    /// ### Parameters
+    /// - `locked`: The locked amount associated with the operation.
+    #[must_use]
+    pub fn locked(mut self, locked: u64) -> Self {
```

Comment:
```suggestion
    /// Locks the specified amount for the `StakeEvent`.
    ///
    /// ### Parameters
    /// - `locked`: The amount to be locked.
    #[must_use]
    pub fn lock(mut self, locked: u64) -> Self {
```

Reply:
I found your suggested comment a bit misleading (at least for me).
Looks like that the method will move a part of the amount field to the locked field, when instead this method actually sets the amount that was locked by the stake contract

Reply:
Wait, maybe we are not on the same page.

When we manipulate the stake amount, we use the suggested approach like the following

```rust
        if to_slash > 0 {
            stake_amount.lock_amount(to_slash);
        }
```

where `lock_amount` is
```rust
    /// Move `amount` to locked value
    pub fn lock_amount(&mut self, amount: u64) {
        self.value -= amount;
        self.locked += amount;
    }

```

However, the code you commented is related to the StakeEvent emitted by the contract when topping up a stake. 
```
                let value = value - locked;
                amount.locked += locked;
                amount.value += value;
                StakeEvent::new(*keys, value).locked(locked)
```


The 2 use cases are bit different.
I can use the lock_amount on the latter too, but it will be cumbersome to derive the values to set in the event (imo)